src/main/java/de/tum/in/www1/artemis/service/connectors/GitService.java

```diff
@@ -375,6 +376,39 @@ public Repository getOrCheckoutRepository(VcsRepositoryUrl repoUrl, String targe
         return getOrCheckoutRepository(repoUrl, localPath, pullOnGet);
     }
 
+    /**
+     * Get the local repository for a given remote repository URL. If the local repo does not exist yet, it will be checked out.
```

Comment:
```suggestion
     * Get the local repository for a given remote repository URL.
     * <p>
     * If the local repo does not exist yet, it will be checked out.
```